helix-term/src/commands.rs

```diff
@@ -916,6 +918,34 @@ fn extend_char_right(cx: &mut Context) {
     doc.set_selection(view.id, selection);
 }
 
+fn extend_char_up(cx: &mut Context) {
+    let count = cx.count();
+    let (view, doc) = current!(cx.editor);
+    let text = doc.text().slice(..);
+    let old_selection = doc.selection(view.id);
+    let mut selection = old_selection.clone();
+    for r in old_selection.iter() {
```

Comment:
I think you'll want to collect a new set of ranges instead, then call `Selection::new`. `push` is very expensive because it will re-order and normalize the selection set on each call, it's intended for those rare cases where you add a single new range.